WalletWasabi/Models/SmartTransaction.cs

```diff
@@ -89,11 +91,40 @@ public SmartTransaction(Transaction transaction, Height height, uint256 blockHas
 
 		#endregion Constructors
 
-		public void SetHeight(Height height, uint256 blockHash = null, int blockIndex = 0)
+		/// <summary>
+		/// Update the transaction with the data acquired from another transaction. (For example merge their labels.)
+		/// </summary>
+		/// <param name="canUnconfirm">By default confirmed transaction won't get unconfirmed. You can override this option.</param>
+		public void Update(SmartTransaction tx, bool canUnconfirm = false)
 		{
-			Height = height;
-			BlockHash = blockHash;
-			BlockIndex = blockIndex;
+			// If this is not the same tx, then don't update.
+			if (this != tx)
```

Comment:
This can be only possible if the caller is doing something wrong, but I don't see it.

Reply:
?

Reply:
This can only happen in case the caller called `Update` with a different `SmartTransaction`. For example:

```c#
var st1 = BuildTransaction(....);
var st2 = BuildADifferentTransaction();
st1.Update(st2);
```

Here, the caller is trying to update `st1` with a different transaction `st2`, what makes no sense. So, only in a case like this one where the caller is doing something wrong the `if` statement makes sense.